Add `enable-disk-spill-encrypt.md` to decrypt the usage of the `spilled-file-encryption-method` (#3978)
pingcap/docs: enable-disk-spill-encrypt.md

---

> title: Enable Disk Spill Encrypt

Reviewer: ```suggestion
title: Enable Encryption for Disk Spill
```

> summary: Learn how to enable disk spill encrypt for TiDB.

Reviewer: ```suggestion
summary: Learn how to enable encryption for disk spill in TiDB.
```

---

> # Enable disk spill encrypt for TiDB

Reviewer: ```suggestion
# Enable Encryption for Disk Spill
```

> When the configuration item `oom-use-tmp-storage` is `true`, if the memory usage of a single SQL statement exceeds the limit of `mem-quota-query` setting, some operators can save the intermediate results during execution as a temporary file to the disk and delete them after the query is completed.

Reviewer: ```suggestion
When the `oom-use-tmp-storage` configuration item is set to `true`, if the memory usage of a single SQL statement exceeds the limit of `mem-quota-query` setting, some operators can save the intermediate results during execution as a temporary file to the disk and delete the file after the query is completed.
```

> Users can enable the disk spill encrypt to prevent attackers from accessing data by reading the temporary files.

Reviewer: ```suggestion
You can enable encryption for disk spill to prevent attackers from accessing data by reading these temporary files.
```

## Configure

> To enable the encryption of the disk spill files, we can configure the item [`spilled-file-encryption-method`](/tidb-configuration-file.md#spilled-file-encryption-method) in the `[security]` section of the TiDB configuration file

Reviewer: ```suggestion
To enable encryption for the disk spill files, you can configure the item [`spilled-file-encryption-method`](/tidb-configuration-file.md#spilled-file-encryption-method) in the `[security]` section of the TiDB configuration file.
```

```toml
[security]
spilled-file-encryption-method = "aes128-ctr"
```

> Possible values for `spilled-file-encryption-method` are `aes128-ctr` and `plaintext`. The default value is `plaintext`, which means encryption is disable.

Reviewer: ```suggestion
Value options for `spilled-file-encryption-method` are `aes128-ctr` and `plaintext`. The default value is `plaintext`, which means that encryption is disabled.
```
